src/app/component/eco-news/components/news-list/news-list.component.ts

```diff
@@ -55,46 +54,43 @@ export class NewsListComponent implements OnInit, OnDestroy {
       .subscribe((data) => this.isLoggedIn = data && data.userId);
   }
 
-  private fetchAllEcoNews(): void {
-    this.ecoNewsSubscription = this.ecoNewsService
-      .newsListSubject
-      .subscribe(this.setAllAndStartingElems.bind(this));
-  }
-
   public onScroll(): void {
     this.addElemsToCurrentList();
   }
 
   private addElemsToCurrentList(): void {
-    const loadingLength = this.allEcoNews.length - this.elements.length > 11 ? 11 :
-      this.allEcoNews.length - this.elements.length;
-
-    this.allEcoNews.forEach((element, index) => {
-      if (index >= this.iterator && index - this.iterator < loadingLength) {
-        this.elements[index] = element;
-      }
-    });
-
-    this.iterator = this.elements.length;
+    if (this.gridOutput) {
+      this.ecoNewsSubscription = this.ecoNewsService.getNewsListByTags(this.iterator++, 12, this.gridOutput)
+        .subscribe(this.setList.bind(this));
+    } else {
+      this.ecoNewsSubscription = this.ecoNewsService.getEcoNewsListByPage(this.iterator++, 12)
+        .subscribe(this.setList.bind(this));
+    }
   }
 
-  private setAllAndStartingElems(data: EcoNewsModel[]): void {
-    this.allEcoNews = [...data];
-    this.elements = data.splice(0, 12);
-    this.iterator = this.elements.length;
-    this.remaining = this.allEcoNews.length;
+  public setList(data): void {
+    this.remaining = data.totalElements;
```

Comment:
There’s a risk that it's a mutable assignment!

Reply:
> There’s a risk that it's a mutable assignment!

data.totalElements is a number, not an array which is vulnerable to it